src/hooks/useDesktopNotifications.ts

```diff
@@ -0,0 +1,144 @@
+import { useEffect, useState } from 'react';
+import { useConfig } from '../contexts/ConfigProvider';
+import { PrData } from '../types';
+import useLogin from './useLogin';
+
+export default function useDesktopNotifications(rawData: PrData[]) {
+  const login = useLogin();
+  const { notificationsEnabled, notifyForComments } = useConfig().config;
+  const [previousData, setPreviousData] = useState<PrData[]>(rawData);
+
+  useEffect(() => {
+    const notifyOfUpdates = () => {
+      if (
+        !login ||
+        !notificationsEnabled ||
+        Notification.permission !== 'granted' ||
+        !previousData.length ||
+        !rawData.length ||
+        previousData.some((p) => p.loading)
+      )
+        return;
+
+      // Notify if any PRs have become ready for review
+      const currentReadyForReview = getPrsReadyForReview(rawData, login);
+      const previousReadyForReview = getPrsReadyForReview(previousData, login);
+      const newPrsReadyForReview = currentReadyForReview
+        .filter(
+          (pr) =>
+            !previousReadyForReview.some(
+              (prevPr) => prevPr.pr.number === pr.pr.number
+            )
+        )
+        .filter((pr) => pr.pr.user?.login !== login);
```

Comment:
nit: This could be a bit cleaner imo:

```tsx
const prevReadyForReview = getPrsReadyForReview(previousData, login);
const prevReadyForReviewNums = prevReadyForReview.map((prevPr) => prevPr.pr.number);
const currentReadyForReview = getPrsReadyForReview(rawData, login);
const newPrsReadyForReview = currentReadyForReview.filter((pr) => !prevReadyForReviewNums.includes(pr.pr.number));
  // I don't think this check will ever be needed as we should only be returning PRs which have you as a reviewer, and it's not possible for the user to be the reviewer
  // .filter((pr) => pr.pr.user?.login !== login);
```

Reply:
This looks better ty

>  I don't think this check will ever be needed as we should only be returning PRs which have you as a reviewer, and it's not possible for the user to be the reviewer
  `.filter((pr) => pr.pr.user?.login !== login);`
  

True, this makes more sense in the `getPrsReadyForReview` function